packages/craco/lib/config.js

```diff
@@ -40,20 +40,43 @@ function processCracoConfig(cracoConfig, context) {
     return applyCracoConfigPlugins(resultingCracoConfig, context);
 }
 
-function loadCracoConfig(context) {
+function getConfigAsObject(context) {
     log("Found craco config file at: ", configFilePath);
 
     const config = require(configFilePath);
     const configAsObject = isFunction(config) ? config(context) : config;
 
     if (!configAsObject) {
-        throw new Error("craco: Config function didn't returned a config object.");
+        throw new Error("craco: Config function didn't return a config object.");
+    }
+    return configAsObject;
+}
+
+function loadCracoConfig(context) {
+    const configAsObject = getConfigAsObject(context);
+
+    if (configAsObject instanceof Promise) {
+        throw new Error(
+            "craco: Config function returned a promise. Use `loadCracoConfigAsync` instead of `loadCracoConfig`."
+        );
+    }
+
+    return processCracoConfig(configAsObject, context);
+}
+
+// Async configs are supported by the "build", "start", and "test" scripts, but not for APIs like createJestConfig.
```

Comment:
Wondering what you mean here when state that it doesn't work for APIs like `createJestConfig`.

Of course you can't called call directly `loadCracoConfigAsync` since the function is not exposed to the consumers as you can't call `loadCracoConfig`.

However if the consumer API code await when loading the async config it should work. Something like:

```
const package = require("./package.json");

require(`./${package.cracoConfig}`).then(cracoConfig => {
    const jestConfig = createJestConfig(cracoConfig, {}, { verbose: true });

    console.log(JSON.stringify(jestConfig, null, 2));
});
```

I'm I wrong?

Reply:
You're absolutely right. My comment here is misleading: I'll update it to be clearer and not mention API calls at all, maybe something like:

> The "build", "start", and "test" scripts use this to wait for any promises to resolve before they run.

Reply:
Sounds good! Thanks